File: src/main/java/com/uiptv/util/TextParserService.java
package com.uiptv.util;

import com.uiptv.model.Account;

import java.util.List;

/**
 * Main service for parsing text into accounts.
 * This class now acts as a factory to select and create the appropriate parser.
 */
public class TextParserService {

    public static final String MODE_STALKER = "Stalker Portal";
    public static final String MODE_XTREME = "Xtreme";
    public static final String MODE_M3U = "M3U Playlists";

    public static List<Account> saveBulkAccounts(String text, String mode, boolean groupAccountsByMac, boolean convertM3uToXtreme) {
        AccountParser parser;

        if (MODE_STALKER.equals(mode)) {
            parser = new StalkerPortalParser();
        } else if (MODE_XTREME.equals(mode)) {
            parser = new XtremeParser();
        } else {
            parser = new M3uParser();
        }

        return parser.parseAndSave(text, groupAccountsByMac, convertM3uToXtreme);
    }
}

File: server/src/main/kotlin/com/uiptv/util/TextParserService.kt
package com.uiptv.util

import com.uiptv.api.AccountParser
import com.uiptv.model.Account

object TextParserService {
    const val MODE_STALKER: String = "Stalker Portal"
    const val MODE_XTREME: String = "Xtreme"
    const val MODE_M3U: String = "M3U Playlists"

    @JvmStatic
    fun saveBulkAccounts(
        text: String,
        mode: String,
        groupAccountsByMac: Boolean,
        convertM3uToXtreme: Boolean
    ): List<Account> {
        val parser: AccountParser = when (mode) {
            MODE_STALKER -> StalkerPortalParser()
            MODE_XTREME -> XtremeParser()
            else -> M3uParser()
        }
        return parser.parseAndSave(text, groupAccountsByMac, convertM3uToXtreme)
    }
}
